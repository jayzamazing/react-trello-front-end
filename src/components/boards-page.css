.boards {
  margin-top: 50px;
}
