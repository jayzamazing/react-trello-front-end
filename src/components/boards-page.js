import React from 'react';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import BoardsForm from './boards-form';
import './boards-page.css';

export function BoardsPage(props) {
    // Only visible to logged in users
    if (!props.loggedIn) {
        return <Redirect to="/" />;
    }
    return (
        <div className="boards container">
          <div className="row">
            <div className="boards-area">
              <h3>Personal Boards</h3>
              <BoardsForm />
            </div>
          </div>
        </div>
    );
}

const mapStateToProps = state => ({
    loggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(BoardsPage);
