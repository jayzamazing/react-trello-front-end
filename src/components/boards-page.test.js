'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import {BoardsPage} from './boards-page';
import {Redirect} from 'react-router-dom';
import toJson from 'enzyme-to-json';

describe('boards-page', () => {
  it('should render page when authenticated', () => {
    const wrapper = shallow(<BoardsPage loggedIn={true}/>);
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });
  it('authenticated snapshot', () => {
    const wrapper = shallow(<BoardsPage loggedIn={true}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders redirect when user is not authenticated', () => {
    const wrapper = shallow(<BoardsPage loggedIn={false}/>);
    expect(wrapper.find(Redirect)).toHaveLength(1);
  });
  it('not authenticated snapshot', () => {
    const wrapper = shallow(<BoardsPage loggedIn={false}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
