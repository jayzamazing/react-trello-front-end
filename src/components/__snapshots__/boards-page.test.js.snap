// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boards-page authenticated snapshot 1`] = `
<div
  className="boards container-fluid"
>
  <div
    className="row"
  >
    <div
      className="boards-area col-xs-12"
    >
      <h3>
        Personal Boards
      </h3>
      <withRouter(Connect(Boards)) />
    </div>
  </div>
</div>
`;

exports[`boards-page not authenticated snapshot 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;
